### DISCUSSION TITLE
一个开源生态的数据模拟仿真器

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#83] [2️⃣ 聊项目]
随着 OpenRank 在 X-lab org 仓库上的落地，项目活跃度/影响力，以及个人活跃度/影响力排名可视化出来，大家开始关心 OpenRank 背后的原理，特别是如何采取行动，以提高自己的影响力，拿到等多的积分换奖金。

![216949c3afa25f07950967957875c1e](https://user-images.githubusercontent.com/15010826/172748419-5cea3511-bd78-42c0-82f7-bd1d9e09ebe1.png)

<img width="1212" alt="a04d2304d1ca5e0371c6da6696dfd6a" src="https://user-images.githubusercontent.com/15010826/172748458-b7a96f4c-dbae-42b1-a024-ce9a86a72ab6.png">

这里实际上是可以做一个数据模拟仿真器的，即“当我投入到某个项目上的某个行为时”，“我的哪个数值”会如何变化？这样做有几个好处：

- **教育目的**：直观的给大家展示 OpenRank 的价值导向，多活跃、多PR、多协作、多群体协作、多在重要项目上协作、等等，能够可视化观察到一些因果关系
- **资源投放策略**：可以做推演，OSPO 有一定的资源，究竟投放到哪些项目上的哪些行为上，收获的效益能够最大，对上汇报争取资源很管用
- **运营优化策略**：动态调整评价指标和参数，引导正确的行为，可以提前进行推演，例如增加什么行为会导致什么结果等

举例：
- 其它行为不变的情况下，增加 A 仓库上 10 个单独 PR 数据，对指标有什么影响；
- 其它行为不变的情况下，增加 A 仓库上 10 个协作 PR 数据（随机协作），对指标有什么影响；
- 其它行为不变的情况下，增加 A 仓库上 10 个单独 PR 数据（特定选择有影响的贡献者），对指标有什么影响；
- 投入 100 个 PR，应该在哪些项目上，使得整个项目的影响力总量提升最大
- 其它数据不变的情况下，投入多少个 PR 到项目 B 上，B 的排名能够提升 10 名
- 我想战略培育项目 C，怎样设计一些参数和权重，让大家知道，同样贡献一个 PR，在项目 C 上贡献，价值最大；
- 我想培育项目 D 的生态影响力，应该鼓励谁去哪些项目上贡献，取得的成效会更好
- 等等可以有很多，有些很难，有些也行

![image](https://user-images.githubusercontent.com/15010826/172752382-1c744112-a5b9-4b6a-bf8e-b287aa6c8990.png)
（没有找到好的图，先将就着用吧）

**Comment by frank-zsy** [comment]
王老师提的是非常好的科研点，之前我总是想着用同一份实际数据集，使用不同的参数集来做对比。但事实上可以通过修改数据集的方式来比较会更加直观。

这里面添加新的数据都需要考虑关联，例如可以随机找已有贡献者添加关联，随机找本来没有关联的外部开发者添加关联，找外部的同领域知名开发者添加关联，找外部的非当前领域的知名开发者添加关联来观察数据变化等，都是非常有意思的。而且很适合放在论文里。

**Reply by frank-zsy** [reply]
而且这可以很好的体现活跃度和影响力的区别，因为无论找什么样的开发者增加活跃，统计上都是类似甚至相同的，但在网络指标上就会有较大的差异。

**Comment by will-ww** [comment]
是的，这块一直在想是个什么形态，看了大家讨论，慢慢有点清晰了~
